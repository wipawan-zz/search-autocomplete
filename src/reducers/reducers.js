const initialState = {
  query: "",
  results: [],
  selectedItem: null,
  error: null
};

export default function reducers(state = initialState, action) {
  switch(action.type) {
    case 'CHANGE_QUERY':
      return {
        ...state,
        query: action.query
      };
    case 'FETCH_REPOS_SUCCESS':
      return {
        ...state,
        results: action.results
      };
    case 'CLEAR_AUTOCOMPLETE':
      return {
        ...state,
        results: []
      };
    case 'SELECT_ITEM':
      return {
        ...state,
        selectedItem: action.selectedItem
      };
    case 'API_CALL_FAILURE':
      return {
        ...state,
        error: action.error
      }
    default:
      return state;
  }
}