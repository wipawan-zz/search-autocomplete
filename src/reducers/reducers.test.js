import reducers from './reducers';

describe('reducers', () => {
  describe('initialState', () => {
    test('is correct', () => {
      const action = {
        type: 'DUMMY_ACTION'
      };
      expect(reducers(undefined, action)).toMatchSnapshot();
    });
  });

  describe('change query', () => {
    test('returns correct query', () => {
      const action = {
        type: "CHANGE_QUERY",
        query: "q"
      };
      expect(reducers(undefined, action)).toMatchSnapshot();
    });
  });
  
  describe('fetch repos success', () => {
    test('updates results after successfully fetch repos', () => {
      const action = {
        type: "FETCH_REPOS_SUCCESS",
        results: [
          "dummy_items"
        ]
      };
      expect(reducers(undefined, action)).toMatchSnapshot();
    });
  });

  describe('clear autocomplete', () => {
    test('empties results array in the store', () => {
      const action = {
        type: "CLEAR_AUTOCOMPLETE",
      };
      expect(reducers(undefined, action)).toMatchSnapshot();
    });
  });

  describe('select item', () => {
    test('select an item', () => {
      const action = {
        type: "SELECT_ITEM",
        selectedItem: {
          "item": "dummyItem"
        }
      };
      expect(reducers(undefined, action)).toMatchSnapshot();
    });
  });
});