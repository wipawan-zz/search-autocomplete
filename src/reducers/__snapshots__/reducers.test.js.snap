// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers change query returns correct query 1`] = `
Object {
  "error": null,
  "query": "q",
  "results": Array [],
  "selectedItem": null,
}
`;

exports[`reducers clear autocomplete empties results array in the store 1`] = `
Object {
  "error": null,
  "query": "",
  "results": Array [],
  "selectedItem": null,
}
`;

exports[`reducers fetch repos success updates results after successfully fetch repos 1`] = `
Object {
  "error": null,
  "query": "",
  "results": Array [
    "dummy_items",
  ],
  "selectedItem": null,
}
`;

exports[`reducers initialState is correct 1`] = `
Object {
  "error": null,
  "query": "",
  "results": Array [],
  "selectedItem": null,
}
`;

exports[`reducers select item select an item 1`] = `
Object {
  "error": null,
  "query": "",
  "results": Array [],
  "selectedItem": Object {
    "item": "dummyItem",
  },
}
`;
